README.md

```diff
@@ -0,0 +1,105 @@
+
+- Considering the extensive code space the Linux kernel deals with, it provides a compile-time analysis instead of a static analysis of the commit history of the Linux kernel, presenting changes only in files used during compilation. 
+- Generating not only web source files but also lists of all source files and dependencies/header files used in kernel compilation, facilitating additional analysis purposes. (More details in [Files Generated](#files-generated))
+- Enabling comparison between two specific tags/releases in the Linux kernel, highlighting all newly added and deleted lines. This provides a clear layout of differences between the tags. While Cregit organizes information by files and embeds the latest commit details in each line/token, it does not support direct comparison between two tags.
+- User customization: allows users to define the URL of the Linux root repository and specify the subsystem for analysis. (More details in [How to Use](#how-to-use))
+
+## How to use
+
+To utilize this tool in your Linux environment (compatible with Ubuntu and Debian), follow these steps:
+
+Clone the repository:
+
+```bash
+git clone <repository_url>
+```
+
+Navigate to the cloned repository:
+
+```bash
+cd <repository_directory>
+```
+
+Execute the tool by specifying the old and new tags:
+
+```bash
+./run_tool.sh <tag1> <tag2> [-c clone_path] [-u repo_link] [-s subsystem]
+```
+- `<tag1>`: Specifies the old version tag.
+- `<tag2>`: Specifies the new version tag.
+- `-c <clone_path>`: Optional. Defines the user-specified path to clone the Linux source code repository.
+- `-u <repo_link>`: Optional. Provides the URL for the Linux source code repository.
+- `-s <subsystem>`: Optional. Specifies the subsystem for analysis (e.g., -s security).
+
+## Files Generated
+
+**/build_data:**
+
+- `sourcefile.txt` - List of all built source code files
+- `headerfile.txt` - List of all built dependency files
+- `git_diff_sourcefile.txt` - Git diff report for source code files
+- `git_diff_headerfile.txt` - Git diff report for dependency files
+- `tokenize_header.json` - Metadata for commit git diff for dependency files
+- `tokenize_source.json` - Metadata for commit git diff for source files
+
+**/web_source_codes:**
+
+- `index.html` - Click on to view the result
+
+## Structure of the Tool
+
+The tool operates through a structured process to generate a comprehensive change impact analysis report. Here's a detailed breakdown of its operation:
+
+### I. Compilation File List Generation
+
+#### Header File
+
+During linux kernel compilation, `Makefile.build` calls `$K/scripts/basic/fixdep.c` to generate a .cmd file for each source that collects dependency information during compilation.
+
+This tool incorporates a modification that applies a patch (`patch.file`) to `scripts/basic/fixdep.c`, enabling it to output dependency information into a **list of header files** when building the kernel.
```

Comment:
It might help to expand on why this patch is necessary?

Is something currently broken in fixdep.c? Or did we need to add something to get additional information that we needed?

Reply:
fixdep.c generates a .cmd file for each source file. The patch collects all dependencies from each .cmd file and outputs a dependency_list.txt file, which contains the compile-time dependency file list after kernel compilation. I have added this explanation to the new README.md